src/snowflake/cli/plugins/snowpark/package/packages_in_snowflake.py

```diff
@@ -171,3 +148,70 @@ def write_requirements_file_in_snowflake_format(
 
         if formatted_requirements:
             file_path.write_text("\n".join(formatted_requirements))
+
+
+class PackagesAvailableInSnowflakeManager(SqlExecutionMixin):
+    _snowflake_channel_url: str = (
+        "https://repo.anaconda.com/pkgs/snowflake/channeldata.json"
+    )
+
+    # TODO in v3.0: Keep only SQL query, remove fallback to JSON with channel's metadata
+    def find_packages_available_in_snowflake(self) -> PackagesAvailableInSnowflake:
+        """
+        Finds python packages available in Snowflake to use in functions and stored procedures.
+        It tries to get the list of packages using SQL query
+        but if the try fails then the fallback is to parse JSON containing info about Snowflake's Anaconda channel.
+        """
+        try:
+            packages = self._query_snowflake_for_available_packages()
+        except Exception as ex:
+            log.warning(
+                "Cannot fetch available packages information from Snowflake. "
+                "Please check your connection configuration. "
+                "Fallback to Anaconda channel metadata."
+            )
+            log.debug("Available packages query failure: %s", ex.__str__(), exc_info=ex)
+            packages = self._get_available_packages_from_anaconda_channel_info()
+        return PackagesAvailableInSnowflake(packages)
+
+    def _query_snowflake_for_available_packages(self) -> dict[str, AvailablePackage]:
+        cursor = self._execute_query(
+            "select package_name, version from information_schema.packages where language = 'python'"
+        )
+        if cursor.rowcount is None or cursor.rowcount == 0:
+            raise SnowflakeSQLExecutionError()
+        packages: dict[str, AvailablePackage] = {}
+        for row in cursor:
+            if not (package_name := row[0]):
```

Comment:
Have you considered executing the query with `cursor_class=DictCursor`? In this way we don't have to operate on indexes

Reply:
Ok, I used DictCursor